packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -247,42 +255,62 @@ export async function scanParentDirs(
     readPackageJson && pkgJsonPath
       ? JSON.parse(await fs.readFile(pkgJsonPath, 'utf8'))
       : undefined;
-  const [yarnLockPath, npmLockPath, pnpmLockPath] = await walkParentDirsMulti({
-    base: '/',
-    start: destPath,
-    filenames: ['yarn.lock', 'package-lock.json', 'pnpm-lock.yaml'],
-  });
+
   let lockfilePath: string | undefined;
   let lockfileVersion: number | undefined;
-  let cliType: CliType = 'yarn';
-
-  const [hasYarnLock, packageLockJson, pnpmLockYaml] = await Promise.all([
-    Boolean(yarnLockPath),
-    npmLockPath
-      ? readConfigFile<{ lockfileVersion: number }>(npmLockPath)
-      : null,
-    pnpmLockPath
-      ? readConfigFile<{ lockfileVersion: number }>(pnpmLockPath)
-      : null,
-  ]);
-
-  // Priority order is Yarn > pnpm > npm
-  if (hasYarnLock) {
-    cliType = 'yarn';
-    lockfilePath = yarnLockPath;
-  } else if (pnpmLockYaml) {
-    cliType = 'pnpm';
-    lockfilePath = pnpmLockPath;
-    lockfileVersion = Number(pnpmLockYaml.lockfileVersion);
-  } else if (packageLockJson) {
-    cliType = 'npm';
-    lockfilePath = npmLockPath;
-    lockfileVersion = packageLockJson.lockfileVersion;
+  let cliType: CliType | undefined;
+
+  if (spawnOpts?.env?.ENABLE_EXPERIMENTAL_COREPACK === '1' && readPackageJson) {
+    const packageManager = packageJson?.packageManager?.split('@')[0]; // { "packageManager": "pnpm@8.0.0" } -> "pnpm"
```

Comment:
Another problem: the `packageJson` here is from `walkParentDirs()` but finding the nearest package.json is not enough. It has to be the nearest one that contains `packageManager` prop. So we'll likely need to add a predicate to `walkParentDirs()`